packages/material-ui/src/styles/withStyles.js

```diff
@@ -219,53 +278,7 @@ const withStyles = (stylesOrCreator, options = {}) => Component => {
     unsubscribeId = null;
 
     render() {
-      const { classes: classesProp, innerRef, ...other } = this.props;
-
-      let classes;
-      let renderedClasses = {};
-
-      if (!this.disableStylesGeneration) {
-        const sheetManager = this.sheetsManager.get(this.stylesCreatorSaved);
-        const sheetsManagerTheme = sheetManager.get(this.theme);
-        renderedClasses = sheetsManagerTheme.sheet.classes;
-      }
-
-      if (classesProp) {
-        classes = {
-          ...renderedClasses,
-          ...Object.keys(classesProp).reduce((accumulator, key) => {
-            warning(
-              renderedClasses[key] || this.disableStylesGeneration,
-              [
-                `Material-UI: the key \`${key}\` ` +
-                  `provided to the classes property is not implemented in ${getDisplayName(
-                    Component,
-                  )}.`,
-                `You can only override one of the following: ${Object.keys(renderedClasses).join(
-                  ',',
-                )}`,
-              ].join('\n'),
-            );
-
-            warning(
-              !classesProp[key] || typeof classesProp[key] === 'string',
-              [
-                `Material-UI: the key \`${key}\` ` +
-                  `provided to the classes property is not valid for ${getDisplayName(Component)}.`,
-                `You need to provide a non empty string instead of: ${classesProp[key]}.`,
-              ].join('\n'),
-            );
-
-            if (classesProp[key]) {
-              accumulator[key] = `${renderedClasses[key]} ${classesProp[key]}`;
-            }
-
-            return accumulator;
-          }, {}),
-        };
-      } else {
-        classes = renderedClasses;
-      }
```

Comment:
All of this logic was just moved to a method for readability. It was originally going to be used in the constructor too, hence the method, but it probably doesn't have to be there anymore. The readability is still nice.